### PR TITLE
fix(javascript-sdk): remove-support-property

### PR DESCRIPTION
**Pull request by ryanbas21**
BREAKING CHANGE: remove the ability to use 'modern' as a means of making calls. fetch complicates the auth process too much for the value it provides. we've opted to remove this support property, automatically defaulting to the iframe.

# JIRA Ticket

https://bugster.forgerock.org/jira/browse/SDKS-2330

# Description
remove the support property and remove tests that test, the modern fetch call. 

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
e2e's and not using modern at all.

# Definition of Done

Check all that apply

- [x] Acceptance criteria is met.
- [x] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: e2e/autoscript-suites/src/suites/config-request-middleware.test.ts [hunk 1]
```diff
@@ -44,23 +44,6 @@ test.describe('Test request middleware with login flow', () => {
     expect(messageArray.includes('User info successfully responded')).toBe(true);
     expect(messageArray.includes('Logout successful')).toBe(true);
 
-    // Test for absence of error logs for FRUser.logout
-    expect(messageArray.includes('Session logout was not successful')).toBe(false);
-    expect(messageArray.includes('OAuth endSession was not successful')).toBe(false);
-    expect(messageArray.includes('OAuth revokeToken was not successful')).toBe(false);
```
Comment (cerebrl) [inline e2e/autoscript-suites/src/suites/config-request-middleware.test.ts:50]: Why were these lines removed?

**Review by cerebrl — COMMENTED** [review]
Looks good. Just had one question.

File: e2e/autoscript-suites/src/suites/config-request-middleware.test.ts (hunk 1 above)
Comment (ryanbas21) [reply e2e/autoscript-suites/src/suites/config-request-middleware.test.ts:50]: hmm looks like i got greedy let me double check.

Comment (ryanbas21) [reply e2e/autoscript-suites/src/suites/config-request-middleware.test.ts:50]: looks like just the diff was moved?

Comment (cerebrl) [reply e2e/autoscript-suites/src/suites/config-request-middleware.test.ts:50]: Oh, okay, I see it. Looks good.
